File: domain/src/main/java/com/insurtech/kanguro/domain/model/StatusDescriptionType.kt
package com.insurtech.kanguro.domain.model

enum class StatusDescriptionType(val stringValue: String) {
    Info("Info"),
    Warn("Warn"),
    Error("Error"),
    Unknown("Unknown");

    companion object {
        fun fromString(value: String): StatusDescriptionType {
            return StatusDescriptionType.values().firstOrNull { it.stringValue == value } ?: Unknown
        }
    }
}

File: iosApp/Modules/KanguroPet/KanguroPetDomain/Sources/KanguroPetDomain/Models/StatusDescriptionType.swift
public enum StatusDescriptionType: String, Equatable, Codable {
    case Info
    case Warn
    case Error
    case Unknown
}

public extension StatusDescriptionType {
    init(value: String) {
        switch value.lowercased() {
        case "info": self = .Info
        case "warn": self = .Warn
        case "error" : self = .Error
        default: self = .Unknown
        }
    }
}
